All I know for sure is that it's repeatable on my Win10 machine.

I'm thinking that it's not finding the TMC26xx files when https://github.com/trinamic/TMC26XStepper/archive/master.zip is enabled and https://github.com/trinamic/TMC26XStepper.git is commented out.

I get the same compile errors in steps 4 & 7 if I do the following:
1. uncomment https://github.com/trinamic/TMC26XStepper/archive/master.zip and comment out.https://github.com/trinamic/TMC26XStepper.git
2. PIO Clean(Due)
3. Delete the .piolibdeps directory
4. PIO Build(Due)
5. PIO Clean(Due)
6. Delete the contents of the directory .piolibdeps/master@src-d7635cbe3386c6c3bf6c9ab5b4141aa8 but leave the directory there
7. PIO Build(Due)

The contents of the directory .piolibdeps/master@src-d7635cbe3386c6c3bf6c9ab5b4141aa8 is NOT restored in steps 7 and 8.  

If I try to do the same thing with https://github.com/trinamic/TMC26XStepper.git enabled then the contents of  .piolibdeps\TMC26XStepper are restored.

----

Here's the error messages I get.
```
Archiving .pioenvs\DUE\lib371\libTMC2208Stepper.a
Compiling .pioenvs\DUE\lib21f\master\TMC26XStepper.cpp.o
.piolibdeps\master\TMC26XStepper.cpp: In member function 'char TMC26XStepper::getStallGuardThreshold()':
.piolibdeps\master\TMC26XStepper.cpp:375:23: error: '_BV' was not declared in this scope
if (result & _BV(6)) {
^

.piolibdeps\master\TMC26XStepper.cpp: In member function 'void TMC26XStepper::send262(long unsigned int)':
.piolibdeps\master\TMC26XStepper.cpp:956:30: error: 'SPCR' was not declared in this scope
unsigned char oldMode =  SPCR & SPI_MODE_MASK;
^
.piolibdeps\master\TMC26XStepper.cpp:956:37: error: 'SPI_MODE_MASK' was not declared in this scope
unsigned char oldMode =  SPCR & SPI_MODE_MASK;
^
```

--- a/platformio.ini
+++ b/platformio.ini
@@ -23,22 +23,17 @@ libdeps_dir = .piolibdeps
 env_default = megaatmega2560
 
 [common]
+default_src_filter = +<src/*> -<src/config>
+build_flags = -fmax-errors=5
 lib_deps =
   https://github.com/MarlinFirmware/U8glib-HAL/archive/dev.zip
   LiquidCrystal_I2C@1.1.2
   TMC2130Stepper
   https://github.com/teemuatlut/TMC2208Stepper/archive/v0.0.3.zip
   Adafruit NeoPixel@1.1.3
-  https://github.com/lincomatic/LiquidTWI2/archive/master.zip
-#  https://github.com/trinamic/TMC26XStepper/archive/master.zip
-  https://github.com/trinamic/TMC26XStepper.git
-  https://github.com/ameyer/Arduino-L6470/archive/master.zip
-
-default_src_filter = +<src/*> -<src/config>
-
-build_flags = -fmax-errors=5
-  -g
-  -ggdb
+  https://github.com/lincomatic/LiquidTWI2/archive/30aa480.zip
+  https://github.com/ameyer/Arduino-L6470/archive/3cd0993.zip
+  https://github.com/trinamic/TMC26XStepper/archive/c1921b4.zip
 
 #################################
 #                               #
@@ -75,6 +70,30 @@ lib_deps    = ${common.lib_deps}
 src_filter  = ${common.default_src_filter}
 
 #
+# Melzi and clones (ATmega1284p)
+#
+[env:melzi]
+platform     = atmelavr
+framework    = arduino
+board        = sanguino_atmega1284p
+build_flags  = ${common.build_flags}
+upload_speed = 57600
+lib_deps     = ${common.lib_deps}
+src_filter   = ${common.default_src_filter}
+
+#
+# Melzi and clones (Optiboot bootloader)
+#
+[env:melzi_optiboot]
+platform     = atmelavr
+framework    = arduino
+board        = sanguino_atmega1284p
+build_flags  = ${common.build_flags}
+upload_speed = 115200
+lib_deps     = ${common.lib_deps}
+src_filter   = ${common.default_src_filter}
+
+#
 # Due (Atmel SAM3X8E ARM Cortex-M3)
 #
 #  - RAMPS4DUE
@@ -86,7 +105,7 @@ framework   = arduino
 board       = due
 build_flags = ${common.build_flags}
 lib_deps    = ${common.lib_deps}
-lib_ignore  = TMC26XStepper
+lib_ignore  = c1921b4
 src_filter  = ${common.default_src_filter}
 
 #
@@ -141,30 +160,6 @@ debug_server   =
   -speed
   auto
   -noir
-
-#
-# Melzi and clones (ATmega1284p)
-#
-[env:melzi]
-platform     = atmelavr
-framework    = arduino
-board        = sanguino_atmega1284p
-build_flags  = ${common.build_flags}
-upload_speed = 57600
-lib_deps     = ${common.lib_deps}
-src_filter   = ${common.default_src_filter}
-
-#
-# Melzi and clones (Optiboot bootloader)
-#
-[env:melzi_optiboot]
-platform     = atmelavr
-framework    = arduino
-board        = sanguino_atmega1284p
-build_flags  = ${common.build_flags}
-upload_speed = 115200
-lib_deps     = ${common.lib_deps}
-src_filter   = ${common.default_src_filter}
 
 #
 # RAMBo

